runtime/plugin/editorconfig.lua

```diff
@@ -1,11 +1,13 @@
-if vim.g.editorconfig_enable == false or vim.g.editorconfig_enable == 0 then
-  return
-end
-
 local group = vim.api.nvim_create_augroup('editorconfig', {})
 vim.api.nvim_create_autocmd({ 'BufNewFile', 'BufRead', 'BufFilePost' }, {
   group = group,
   callback = function(args)
+    -- Buffer-local enable has higher priority
+    local enable = vim.F.if_nil(vim.b.editorconfig_enable, vim.F.if_nil(vim.g.editorconfig_enable, true))
```

Comment:
related to the discussion about a "new convention" (vs `g:loaded_foo`), could we overload `g:editorconfig` / `b:editorconfig` to be both (1) a general place for the plugin's config or whatever, and (2) if it's `false`, it means the user wants to disable the plugin.

- so `g:foo=false` (and `b:foo=false`) means "I want to disable the foo plugin"
- else, `g:foo` (and `b:foo` where relevant) are a table for config or state (as in `b:editorconfig`)

Reply:
Added as a separate commit to this PR.